x/merkledb/trie_test.go

```diff
@@ -1178,3 +1180,109 @@ func TestTrieViewInvalidChildrenExcept(t *testing.T) {
 	require.True(view3.invalidated)
 	require.Empty(view1.childViews)
 }
+
+
+func Test_Trie_ConcurrentReadWrite(t *testing.T) {
+	trie, err := getBasicDB()
+	require.NoError(t, err)
+	require.NotNil(t, trie)
+	newTrie, err := trie.NewView()
+	require.NoError(t, err)
+
+	var wg sync.WaitGroup
+	defer wg.Wait()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err := newTrie.Insert(context.Background(), []byte("key"), []byte("value"))
+		require.NoError(t, err)
+	}()
+
+	// Wait for insert to complete first
+	time.Sleep(10 * time.Millisecond)
```

Comment:
we should not rely on `time.Sleep` for ordering operations, as this will lead us to flaky tests.
Also if we wait for the completion of the above goroutine then we are not testing concurrent operations.

Could we instead use `require.Eventually` on the result of `GetValue` with an appropriate timeout or use an explicit loop?

Reply:
Refactored to use require.Eventually.